app/services/journals/create_service.rb

```diff
@@ -910,7 +906,7 @@ def same_user?(predecessor)
     end
 
     def same_cause?(predecessor, cause)
-      (predecessor.cause.blank? && cause.blank?) || predecessor.cause == cause
+      (predecessor.cause.blank? && cause.blank?) || predecessor.cause == cause.to_h
```

Comment:
Defining the implicit `to_hash` method on the [cause class](https://github.com/opf/openproject/pull/12882/files#diff-1850d4434c148aa7386b223a7a7589175eaf720b49f94d9a128ea328619ffe4dR38), would automatically do the `to_hash` conversion when it is compared against another hash. 
```suggestion
      (predecessor.cause.blank? && cause.blank?) || predecessor.cause == cause
```